kopf/structs/hierarchies.py

```diff
@@ -8,8 +8,14 @@ def build_object_reference(body):
     """
     Construct an object reference for the events.
     """
+    # apiVersion attribute is not consistent across api resources
+    if 'apiVersion' in body:
```

Comment:
Thank you so much for contributing to this project. 
Would be great to extract this code to a function to reduce the duplication.

Reply:
Done

Reply:
Thanks @olivier-mauras, 
Would also be great if you could add one or more resource names as example for each attribute name.

Reply:
I added a  comment https://github.com/zalando-incubator/kopf/issues/166#issuecomment-519126078 on the bug I opened.
It's actually not the API resources that differ but the body returned by on.xxxx() handlers.